### PR TITLE
Merge `ShelleyLedgerEra` and `CardanoLedgerEra` to a single type family `LedgerEra`

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    Merge `ShelleyLedgerEra` and `CardanoLedgerEra` to a single type family `LedgerEra`
  type:
  # - feature        # introduces a new feature
  - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

We previously defined two separate type families `ShelleyLedgerEra` and `CardanoLedgerEra` but never use `CardanoLedgerEra` anywhere.

`ShelleyLedgerEra` is basically the same as `CardanoLedgerEra` except Byron is missing.

There is no reason to have two type families because `CardanoLedgerEra` works everywhere `ShelleyLedgerEra` currently does.

This PR merges the two type families into the one `LedgerEra` type family (which includes Byron) which shortens the commonly used name.

Doing this opens the possibility of unifying Byron with the other eras for the minimal set of operations we care to maintain for hard forking purposes.

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behaviour, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-api/internal/Cardano/Api/Eon/ShelleyBasedEra.hs [hunk 1]
```diff
@@ -313,25 +311,6 @@ shelleyBasedToCardanoEra ShelleyBasedEraAlonzo  = AlonzoEra
...
--- ledger library which allows writing conversion functions in a more generic
--- way.
---
-type family ShelleyLedgerEra era = ledgerera | ledgerera -> era where
-  ShelleyLedgerEra ShelleyEra = Consensus.StandardShelley
-  ShelleyLedgerEra AllegraEra = Consensus.StandardAllegra
-  ShelleyLedgerEra MaryEra    = Consensus.StandardMary
-  ShelleyLedgerEra AlonzoEra  = Consensus.StandardAlonzo
-  ShelleyLedgerEra BabbageEra = Consensus.StandardBabbage
-  ShelleyLedgerEra ConwayEra  = Consensus.StandardConway
```
Comment (newhoggy) [inline cardano-api/internal/Cardano/Api/Eon/ShelleyBasedEra.hs:333]: Delete this.  Use `LedgerEra` instead.

File: cardano-api/internal/Cardano/Api/Eras/Core.hs [hunk 2]
```diff
@@ -409,15 +409,15 @@ inAnyCardanoEra era a =
...
-  CardanoLedgerEra BabbageEra = L.BabbageEra L.StandardCrypto
-  CardanoLedgerEra ConwayEra  = L.ConwayEra  L.StandardCrypto
+type family LedgerEra era = ledgerera | ledgerera -> era where
+  LedgerEra ByronEra   = L.ByronEra   L.StandardCrypto
+  LedgerEra ShelleyEra = L.ShelleyEra L.StandardCrypto
+  LedgerEra AllegraEra = L.AllegraEra L.StandardCrypto
+  LedgerEra MaryEra    = L.MaryEra    L.StandardCrypto
+  LedgerEra AlonzoEra  = L.AlonzoEra  L.StandardCrypto
+  LedgerEra BabbageEra = L.BabbageEra L.StandardCrypto
+  LedgerEra ConwayEra  = L.ConwayEra  L.StandardCrypto
```
Comment (newhoggy) [inline cardano-api/internal/Cardano/Api/Eras/Core.hs:423]: Rename `CardanoLedgerEra` which is never used to `LedgerEra` which we use everywhere in place of `ShelleyLedgerEra`.

**Review by smelc — APPROVED** [review]
LGTM :+1:

File: cardano-api/internal/Cardano/Api/Eras/Core.hs (hunk 2 above)
Comment (carbolymer) [reply cardano-api/internal/Cardano/Api/Eras/Core.hs:423]: I think the change is pretty harmless, but can you check if anything breaks in `cardano-cli`?

**Review by carbolymer — APPROVED** [review]

File: cardano-api/internal/Cardano/Api/Eras/Core.hs (hunk 2 above)
Comment (newhoggy) [reply cardano-api/internal/Cardano/Api/Eras/Core.hs:423]: Yep.  Already checked.

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
This is not the direction we are taking cardano-api in. We do not care about all the eras, only mainnet and the upcoming era. The first step in this is to factor out the Byron era and only be concerned with shelley based eras. Consolidating the shelley based eras with cardano eras is moving in the wrong direction and prolongs this eventual separation.

**Comment by newhoggy** [comment]
It's worth noting that this PR doesn't commit us to a particular direction.  On it's own it serves as a straightforward simplification. with the only impact on the rest of the code being a rename and removing a type we don't use anywhere.

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
This is an unnecessary change. It is also incorrect because it implies the existence of types in Byron that do no exist. We will be able to simply things further after the Byron related code is isolated.

**Comment by newhoggy** [comment]
What type is Byron type is being implied that doesn't exist?

**Pull request closed without merging**